src/query/allowed_objects.c

```diff
@@ -14,18 +14,49 @@ static const char *const g_allowed_casts[] = {
     "ftod", "dtof",
     "int4_numeric", "float4_numeric", "float8_numeric",
     "numeric_float4", "numeric_float8",
+    "date_timestamptz",
     /**/
 };
 
-static const char *const g_allowed_builtins[] = {
+typedef struct FunctionWithName
+{
+  const char *name;
+  int primary_arg;
+} FunctionWithName;
+
+typedef struct FunctionWithOid
+{
+  Oid funcid;
+  int primary_arg;
+} FunctionWithOid;
```

Comment:
Can we merge these 2 structs and initialize missing fields during oid cache build?

I.e. we can have `("some_func", InvalidOid, 0)` OR `(NULL, F_NUMERIC_ROUND_INT, 0)`. 
During init, we fill out the 0-OID from the Name or we fill out the NULL-Name from OID (the latter might not be necessary?).

Would this simplify the lookup logic? Also would get rid of the func name lookups we do during validation.

Reply:
The part "initialize during oid cache build" is the runtime population we were originally avoiding here. If we change the preferred approach to allow runtime population, this sounds like a viable alternative to the `Oid -> FunctionInfo` mapping I had in mind